src/app/posts/post/post.component.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { Subject, takeUntil } from 'rxjs';
+import { PostsDataService } from '../shared/services/posts-data.service';
+
+@Component({
+  selector: 'app-post',
+  templateUrl: './post.component.html',
+  styleUrls: ['./post.component.scss'],
+  host: { style: 'width:90%; height:70%' },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PostComponent implements OnInit, AfterViewInit, OnDestroy {
+  private unsubscribe$ = new Subject();
+  public isEditable = false;
+  public isEditPage = this.router.url.includes('edit');
+  public canAddNew = this.router.url.includes('add');
+  public postId = Number(this.activatedRoute.snapshot.paramMap.get('id'));
+  public postForm = new FormGroup({
+    id: new FormControl(0),
+    userId: new FormControl(0),
+    title: new FormControl('', Validators.required),
+    body: new FormControl('', Validators.required),
+  });
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private postsDataService: PostsDataService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.canAddNew) {
```

Comment:
That's why you need separate component for Edit and Add, so you don't mix the logic in 1 method